### PR TITLE
[BC-16] Make User main page

### PR DESCRIPTION
**Pull request by CorrectACHU**

### CODE REVIEW

**Comment by CorrectACHU** [comment]
Added logo to the header and logout button.


image
image

Now user can see all his tickets and click on them to move to ticket details page - user will automatically pushed.


image
image

File: FrontEnd/src/stores/store.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    const isLoggedIn =  ref(false)
+    const role = ref('')
+
+    function changeIsLoggedInToTrue() {
+        isLoggedIn.value = true
+    }
+
+    function changeIsLoggedInToFalse() {
+        isLoggedIn.value = false
+    }
```
Comment (rmaskalenka) [inline FrontEnd/src/stores/store.ts]: ummm... why? please use a simple setter instead

File: FrontEnd/src/views/MainPage.vue [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import router from '@/router'
+import { getTokenFromCookie, redirectByPersonRole } from '@/helpers/service'
+import { getRoleFromCookie } from '@/helpers/service'
+
+const personStore = authStore()
+
+if (getTokenFromCookie() === '') {
+  router.push('/login')
+} else {
+  personStore.isLoggedIn = true
```
Comment (rmaskalenka) [inline FrontEnd/src/views/MainPage.vue]: why do we need same flag as in `authStore` but in another store? Looks like an unnecessary data duplication to me

File: FrontEnd/src/views/user/UserMainPage.vue [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import { type CardInGeneral } from '@/stores/modelInterfaces'
+import { getTokenFromCookie } from '@/helpers/service'
+import { authStore } from '@/stores/store'
+import { SERVER_URL } from '@/helpers/envVars'
+
+const personStore = authStore()
+
+const token = getTokenFromCookie()
+const listOfTickets: CardInGeneral[] = reactive([])
+const getListOfTickets = async () => {
```
Comment (rmaskalenka) [inline FrontEnd/src/views/user/UserMainPage.vue]:
```suggestion
async function getListOfTickets() {
```

**Review by rmaskalenka — CHANGES_REQUESTED** [review]
please close the above issues

File: FrontEnd/src/components/AppHeader.vue [hunk 4]
```diff
@@ -1,14 +1,54 @@
...
 <script setup lang="ts">
+import { cleanCookie } from '@/helpers/service'
+import router from '@/router'
+import { authStore } from '@/stores/store'
+const personStore = authStore()
+
+const logout = () => {
+  personStore.changeIsLoggedInToFalse()
+  cleanCookie()
+  router.push('/login')
```
Comment (rmaskalenka) [inline FrontEnd/src/components/AppHeader.vue]: it's better to use named routes as they are more safe in terms of URL path construction

File: FrontEnd/src/components/TicketInList.vue [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  </div>
+</template>
+
+<script setup lang="ts">
+import router from '@/router'
+
+const props = defineProps(['item'])
+
+const moveToTicket = (id: number) => {
+  router.push(`ticket/${id}`)
```
Comment (rmaskalenka) [inline FrontEnd/src/components/TicketInList.vue]: please use a named route. There is a possibility to pass parameters to route as well https://router.vuejs.org/guide/essentials/named-routes.html

**Review by rmaskalenka — APPROVED** [review]

**Pull request merged**